### ISSUE TITLE
Feature request: additional database permissions option, insertOnly

### ISSUE THREAD

**Issue by shamblesides** [smallbets/userbase#208]
In addition to the current two flags, `readOnly` and `resharingAllowed`, I've found myself wanting a third option, where an invited user could insert new documents, but could only update or delete documents that they themselves inserted.

This might be a new property like `insertOnly: true` or it might be some non-boolean value for `readOnly`, like `{ "readOnly": "others" }`

**Comment by j-berman** [comment]
What if you could instead set more granular permissions on each item with a parameter like this:

```
userbase.insertItem({ 
  databaseName,
  item,
  accessControl: { permissions: 'user' }
})
```

Then, only the user can update/delete the item.

Additionally, the `accessControl` param can take an array of users like this:

```
userbase.insertItem({ 
  databaseName,
  item,
  accessControl: { 
    permissions: 'user',
    users: [{ username: 'alice' }, { username: 'bob }]
  }
})
```

Then only the item creator, Alice, or Bob can update/delete the item.

How's that sound?

**Comment by shamblesides** [comment]
This seems like a great way to do it!

**Comment by j-berman** [comment]
Released in userbase-js v2.7.0!

Piggybacked off the code you wrote for attribution 😃
